integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseConvertCryptoFragment.kt

```diff
@@ -118,39 +120,8 @@ class CoinbaseConvertCryptoFragment : Fragment(R.layout.fragment_coinbase_conver
             selectedCoinBaseAccount = account
         }
 
-        convertViewModel.onContinueEvent.observe(viewLifecycleOwner) { pair ->
-            val swapValueErrorType = convertViewModel.checkEnteredAmountValue()
-            if (swapValueErrorType == SwapValueErrorType.NOError) {
-                if (!pair.first && convertViewModel.dashToCrypto.value == true) {
-                    pair.second?.first?.let { fait ->
-                        if ((viewModel.userPreference.lastCoinbaseBalance?.toDouble() ?: 0.0) <fait.toPlainString().toDouble()) {
-                            val placeBuyOrderError = CoinbaseGenericErrorUIModel(
-                                R.string.we_didnt_find_any_assets,
-                                image = R.drawable.ic_info_red,
-                                positiveButtonText = R.string.buy_crypto_on_coinbase,
-                                negativeButtonText = R.string.close
-                            )
-                            safeNavigate(
-                                CoinbaseServicesFragmentDirections.coinbaseServicesToError(
-                                    placeBuyOrderError
-                                )
-                            )
-                        }
-                    }
-                } else {
-                    if (pair.second?.second != null && viewModel.isInputGreaterThanLimit(pair.second?.second!!)) {
-                        showSwapValueErrorView(SwapValueErrorType.UnAuthorizedValue)
-                    } else {
-                        selectedCoinBaseAccount?.let {
-                            pair.second?.first?.let { fait ->
-                                viewModel.swapTrade(fait, it, pair.first)
-                            }
-                        }
-                    }
-                }
-            } else {
-                showSwapValueErrorView(swapValueErrorType)
-            }
+        convertViewModel.onContinueEvent.observe(viewLifecycleOwner) { request ->
+            proceedWithSwap(request)
```

Comment:
We need to call retry on this portion of the code, so better to have it in a separate method.
Also, `onViewCreated` here is a bit bloated already.